src/features/session_replay/aggregate/index.js

```diff
@@ -55,102 +55,93 @@ export class Aggregate extends AggregateBase {
-    )
-
-    if (shouldSetup) {
-      // The SessionEntity class can emit a message indicating the session was cleared and reset (expiry, inactivity). This feature must abort and never resume if that occurs.
-      this.ee.on(SESSION_EVENTS.RESET, () => {
-        this.scheduler.runHarvest()
-        this.abort(ABORT_REASONS.RESET)
-      })
-
-      // The SessionEntity class can emit a message indicating the session was paused (visibility change). This feature must stop recording if that occurs.
-      this.ee.on(SESSION_EVENTS.PAUSE, () => { this.recorder?.stopRecording() })
-      // The SessionEntity class can emit a message indicating the session was resumed (visibility change). This feature must start running again (if already running) if that occurs.
-      this.ee.on(SESSION_EVENTS.RESUME, () => {
-        if (!this.recorder) return
-        // if the mode changed on a different tab, it needs to update this instance to match
-        const { session } = getRuntime(this.agentIdentifier)
-        this.mode = session.state.sessionReplayMode
-        if (!this.initialized || this.mode === MODE.OFF) return
-        this.recorder?.startRecording()
-      })
+    // The SessionEntity class can emit a message indicating the session was cleared and reset (expiry, inactivity). This feature must abort and never resume if that occurs.
+    this.ee.on(SESSION_EVENTS.RESET, () => {
+      this.scheduler.runHarvest()
+      this.abort(ABORT_REASONS.RESET)
+    })
+
+    // The SessionEntity class can emit a message indicating the session was paused (visibility change). This feature must stop recording if that occurs.
+    this.ee.on(SESSION_EVENTS.PAUSE, () => { this.recorder?.stopRecording() })
+    // The SessionEntity class can emit a message indicating the session was resumed (visibility change). This feature must start running again (if already running) if that occurs.
+    this.ee.on(SESSION_EVENTS.RESUME, () => {
+      if (!this.recorder) return
+      // if the mode changed on a different tab, it needs to update this instance to match
+      const { session } = getRuntime(this.agentIdentifier)
+      this.mode = session.state.sessionReplayMode
+      if (!this.initialized || this.mode === MODE.OFF) return
+      this.recorder?.startRecording()
+    })
+
+    this.ee.on(SESSION_EVENTS.UPDATE, (type, data) => {
+      if (!this.recorder || !this.initialized || this.blocked || type !== SESSION_EVENT_TYPES.CROSS_TAB) return
+      if (this.mode !== MODE.OFF && data.sessionReplayMode === MODE.OFF) this.abort(ABORT_REASONS.CROSS_TAB)
+      this.mode = data.sessionReplay
+    })
+
+    // Bespoke logic for blobs endpoint.
+    this.scheduler = new HarvestScheduler('browser/blobs', {
+      onFinished: this.onHarvestFinished.bind(this),
+      retryDelay: this.harvestTimeSeconds,
+      getPayload: this.prepareHarvest.bind(this),
+      raw: true
+    }, this)
+
+    registerHandler('recordReplay', () => {
```

Comment:
# old man shakes fist at cloud #

I am nit picking but these should be in constants somewhere central.

Reply:
works for me

Reply:
68bea1d9